src/app/article/+store/article.effects.ts

```diff
@@ -38,12 +38,19 @@ export class ArticleEffects {
         return this.actions$.pipe(
             ofType(onToggleFavorite),
             exhaustMap((action) => {
-                return this.articlesService.favorite(action.slug).pipe(
-                    map(({ article }) => {
-                        return onToggleFavoriteSuccess(article);
-                    }),
-                    catchError((err) => onToggleFavoriteFailure(err))
-                );
+                if (action.favorited) {
```

Comment:
Just for information: looks like we can use https://www.learnrxjs.io/learn-rxjs/operators/conditional/iif here.